lib/videoroom/meeting.ex

```diff
@@ -0,0 +1,120 @@
+defmodule Videoroom.Meeting do
+  @moduledoc """
+  A Module responsible for handling a room.
+  """
+  use GenServer
+
+  alias Jellyfish.ServerMessage.PeerCrashed
+  alias Jellyfish.ServerMessage.PeerDisconnected
+  alias Jellyfish.ServerMessage.RoomCrashed
+
+  alias Jellyfish.Room
+
+  @room_table :room_table
+
+  # Api
+
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args,
+      name: {:via, Registry, {Videoroom.Registry, args[:name]}}
+    )
+  end
+
+  @spec add_peer(pid()) :: {:ok, binary()} | {:error, binary()}
+  def add_peer(meeting) do
+    GenServer.call(meeting, {:add_peer})
+  end
+
+  # Callbacks
+
+  @impl true
+  def init(args) do
+    client = Jellyfish.Client.new()
+
+    with {:ok, notifier} <- Jellyfish.Notifier.start(),
+         {:ok, room} <- find_or_create_room(client, args) do
+      Process.monitor(notifier)
+
+      {:ok, %{client: client, notifier: notifier, name: args[:name], room_id: room.id}}
+    else
+      {:error, reason} ->
+        raise("Failed to create a meeting, reason: #{inspect(reason)}")
+    end
+  end
+
+  defp find_or_create_room(client, opts) do
+    name = opts[:name]
+
+    with [{^name, room_id}] <- :ets.lookup(@room_table, name),
+         {:ok, room} <- Room.get(client, room_id) do
+      {:ok, room}
+    else
+      _error ->
+        case Room.create(client, max_peers: opts[:max_peers]) do
+          {:ok, room} ->
+            :ets.insert_new(@room_table, {name, room.id})
+            {:ok, room}
+
+          error ->
+            error
+        end
+    end
```

Comment:
1. If there is an ets entry, but the JF room doesn't exist, this code will create a new one and keep the old entry "leaking" the rooms
2. You are ignoring the error, which seems dangerous, not every error means there's no such room
3. This nesting is IMO quite ugly - the else should be extracted to a separate function